test/extensions/filters/http/ext_authz/ext_authz_fuzz_lib.cc

```diff
@@ -0,0 +1,74 @@
+#include "test/extensions/filters/http/ext_authz/ext_authz_fuzz_lib.h"
+
+#include "envoy/config/core/v3/base.pb.h"
+#include "envoy/extensions/filters/http/ext_authz/v3/ext_authz.pb.validate.h"
+
+#include "source/common/network/address_impl.h"
+#include "source/extensions/filters/http/ext_authz/ext_authz.h"
+
+#include "test/extensions/filters/http/ext_authz/ext_authz_fuzz.pb.h"
+#include "test/extensions/filters/http/ext_authz/ext_authz_fuzz.pb.validate.h"
+#include "test/mocks/network/mocks.h"
+
+#include "gmock/gmock.h"
+
+using testing::Return;
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace ExtAuthz {
+
+ReusableFilterFactory::ReusableFilterFactory()
+    : addr_(std::make_shared<Network::Address::PipeInstance>("/test/test.sock")) {
+  connection_.stream_info_.downstream_connection_info_provider_->setRemoteAddress(addr_);
+  connection_.stream_info_.downstream_connection_info_provider_->setLocalAddress(addr_);
+
+  ON_CALL(decoder_callbacks_, connection())
+      .WillByDefault(Return(OptRef<const Network::Connection>{connection_}));
+  ON_CALL(decoder_callbacks_.stream_info_, dynamicMetadata())
+      .WillByDefault(testing::ReturnRef(metadata_));
+}
+
+std::unique_ptr<Filter>
+ReusableFilterFactory::newFilter(FilterConfigSharedPtr config,
+                                 Filters::Common::ExtAuthz::ClientPtr&& client,
+                                 const envoy::config::core::v3::Metadata& metadata) {
+  metadata_ = metadata;
+  std::unique_ptr<Filter> filter = std::make_unique<Filter>(std::move(config), std::move(client));
+  filter->setDecoderFilterCallbacks(decoder_callbacks_);
+  filter->setEncoderFilterCallbacks(encoder_callbacks_);
+
+  return filter;
+}
+
+absl::StatusOr<std::unique_ptr<Filter>> ReusableFuzzerUtil::setup(
+    const envoy::extensions::filters::http::ext_authz::ExtAuthzTestCase& input,
+    Filters::Common::ExtAuthz::ClientPtr client) {
+  try {
+    TestUtility::validate(input);
+  } catch (const EnvoyException& e) {
+    ENVOY_LOG_MISC(debug, "EnvoyException during validation: {}", e.what());
+    return absl::InvalidArgumentError(absl::StrCat("EnvoyException during validation: ", e.what()));
+  }
+
+  // Prepare filter.
+  const envoy::extensions::filters::http::ext_authz::v3::ExtAuthz proto_config = input.config();
+  FilterConfigSharedPtr config;
+
+  try {
+    config = std::make_shared<FilterConfig>(proto_config, *stats_store_.rootScope(),
+                                            "ext_authz_prefix", factory_context_);
```

Comment:
Should "factory_context_" actually be a type envoy::config::bootstrap::v3::Bootstrap.  Just curious how this compiles?

Reply:
This is just copied from [the current fuzzer test](https://github.com/envoyproxy/envoy/blob/22ad8865ef54ccaa239108e0154aedae0f994aab/test/extensions/filters/http/ext_authz/ext_authz_fuzz_test.cc#L94
).